src/lib/OpenEXRCore/internal_ht_common.cpp

```diff
@@ -22,9 +24,13 @@ make_channel_map (
     {
         std::string c_name(channels[i].channel_name);
 
-        if (c_name == "R") { r_index = i; }
-        else if (c_name == "G") { g_index = i; }
-        else if (c_name == "B") { b_index = i; }
+        /* heuristics to determine whether RGB channels are present */
+        std::transform(c_name.begin(), c_name.end(), c_name.begin(),
+                   [](unsigned char c){ return std::tolower(c); });
+
+        if (c_name == "r" || c_name.compare(0, 3, "red") == 0) { r_index = i; }
+        else if (c_name == "g" || c_name.compare(0, 5, "green") == 0) { g_index = i; }
+        else if (c_name == "b" || c_name.compare(0, 4, "blue")== 0) { b_index = i; }
```

Comment:
This will consider `R` as a red channel, but not a channel which has a layer name with a `R` suffix e.g.`diffuse.R`?
Also, if the image happens to have both a `R` and a `r` channel, it'll pick the `r` channel as the red channel not `R`, which is a little unfortunate, since the standard name is `R`

Ideally it should look for `R` as the channel name, and only fall back to using a channel with a non-standard name, or whose name starts with a layer name, if no `R` channel is found.

Reply:
Proposed algorithm:

- look for R, G and B (case sensitive)
- if not found, look for:
  - r, g and b
  - .r, .g and .b suffixes (case insensitive)
  - red, green, blue (case insensitive)

Anything else, like:
 - `-red`, `-green` and `-blue` suffixes (case insensitive)
 - `red*`, `green*` and `blue*` (case insensitive)
 - `-r`, `-g` and `-b` suffixes (case insensitive)

Reply:
What about the [RGBA interface](https://github.com/AcademySoftwareFoundation/openexr/blob/310e3bd3ec3c8be2e123880e1bedad6c8d3862dc/src/lib/OpenEXR/ImfTiledRgbaFile.cpp#L766)? It looks like it simply appends R, G and B to a common prefix. If so, should the encoder look for R, G and B at the end of the channel name?

Reply:
NVM, it looks like the prefix in the RGBA interface is computed with `prefixFromLayerName()`, which ensures a `.` is present.

Reply:
See revised algorithm.